src/tribler/core/components/restapi/rest/utils.py

```diff
@@ -1,8 +1,30 @@
 """
 This file contains some utility methods that are used by the API.
 """
+import sys
+import threading
+from contextlib import contextmanager
+from types import FrameType
+from typing import Dict, Iterator, List, Optional
+
 from tribler.core.components.restapi.rest.rest_endpoint import HTTP_INTERNAL_SERVER_ERROR, RESTResponse
 
+ONE_SECOND = 1
+
+_switch_interval_lock = threading.Lock()
+
+
+@contextmanager
+def switch_interval(value):
+    """ Temporarily change the sys.setswitchinterval value."""
+    with _switch_interval_lock:
+        previous_value = sys.getswitchinterval()
+        try:
+            sys.setswitchinterval(value)
```

Comment:
**nitpick:** Usually the call like this is placed before the `try` block:
```python
change_something()
try:
   ...
finally:
    change_something_back()
```
In the example above, if `change_something()` behaves atomically and raises an exception, that means that "something" hasn't changed, and restoring it is unnecessary.

In our case, `sys.setswitchinterval(value)` also behaves atomically. It can throw an exception if the value has an incorrect type, but in that case, the internal variable wasn't changed, and restoring anything is unnecessary.